### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by notjames**
Trying to get up to speed with Golang again. Decided to work on these. The intention here was to remove the test tables from the code into a separate JSON file, which should allow for simpler tests in the future. Fixed some bugs, etc.

### CODE REVIEW

File: cloud/ssh/actuators/machine/actuator_deployer.go [hunk 1]
```diff
@@ -48,7 +48,7 @@ func (a *Actuator) getPrivateKey(c *clusterv1.Cluster, namespace string, secretN
 
 		secret, err := secretsClient.Get(secretName, meta_v1.GetOptions{})
 		if err != nil {
-			glog.Errorf("could not retrieve machine secret", err)
+      glog.Errorf("could not retrieve machine secret: %v", err)
 			return "", "", err
 		}
```
Comment (oneilcin) [inline cloud/ssh/actuators/machine/actuator_deployer.go]: Good catches on the error msgs!

File: cloud/ssh/actuators/machine/configwatch_test.go [hunk 2]
```diff
@@ -1,64 +1,86 @@
...
+
+  if err != nil {
+    log.Fatal(err)
+  }
+
+  os.Exit(m.Run())
+}
+
+// TestNewConfigWatch add new tests in test-cases/configwatch.json
 func TestNewConfigWatch(t *testing.T) {
```
Comment (oneilcin) [inline cloud/ssh/actuators/machine/configwatch_test.go]: I would prefer that we not move test cases to a json file.  If you look at other golang test cases, they are in the code - as was done previous to this change.  To me the json file test cases make it harder to understand the tests, and then we have to understand how it is parsed.

**Comment by davidewatson** [comment]
IMO splitting the test data from the source complicates the tests because it requires more concepts (e.g. `tc.Type`) and is not as common as far as I have seen.

File: cloud/ssh/actuators/machine/configwatch_test.go (hunk 2 above)
Comment (notjames) [reply cloud/ssh/actuators/machine/configwatch_test.go]: Yeah, as you see, @davidewatson said the same thing, so I'll revert those changes and resubmit.

**Pull request closed without merging**